src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/test_mrpProportionalDerivative.py

```diff
@@ -0,0 +1,177 @@
+    testProc = unitTestSim.CreateNewProcess(unitProcessName)
+    testProc.addTask(unitTestSim.CreateNewTask(unitTaskName, testProcessRate))
+
+    # Construct algorithm and associated C++ container
+    module = mrpProportionalDerivative.MrpProportionalDerivative()
+    module.ModelTag = "mrpPD"
+
+    # Add test module to runtime call list
+    unitTestSim.AddModelToTask(unitTaskName, module)
+
+    # Initialize the test module configuration data
+    module.K = 0.15
+    module.P = 150.0
+    if setExtTorque:
+        module.knownTorquePntB_B = np.array([0.1, 0.2, 0.3])
+
+    #   Create input message and size it because the regular creator of that message
+    #   is not part of the test.
+    #   attGuidOut Message:
+    guidCmdData = messaging.AttGuidMsgPayload()
+    guidCmdData.sigma_BR = np.array([0.3, -0.5, 0.7])
+    guidCmdData.omega_BR_B = np.array([0.010, -0.020, 0.015])
+    guidCmdData.omega_RN_B = np.array([-0.02, -0.01, 0.005])
+    guidCmdData.domega_RN_B = np.array([0.0002, 0.0003, 0.0001])
+    guidInMsg = messaging.AttGuidMsg().write(guidCmdData)
+
+    # vehicleConfig FSW Message:
+    vehicleConfigIn = messaging.VehicleConfigMsgPayload()
+    vehicleConfigIn.ISCPntB_B = [1000., 0., 0.,
+                                  0., 800., 0.,
+                                  0., 0., 800.]
+    vcInMsg = messaging.VehicleConfigMsg().write(vehicleConfigIn)
+
+    # Setup logging on the test module output message so that we get all the writes to it
+    dataLog = module.cmdTorqueOutMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, dataLog)
+
+    # connect messages
+    module.vehConfigInMsg.subscribeTo(vcInMsg)
+    module.guidInMsg.subscribeTo(guidInMsg)
+
+    # Need to call the self-init and cross-init methods
+    unitTestSim.InitializeSimulation()
+
+    # Step the simulation to 3*process rate so 4 total steps including zero
+    unitTestSim.ConfigureStopTime(macros.sec2nano(1.0))  # seconds to stop simulation
+    unitTestSim.ExecuteSimulation()
+
+    trueVector = [findTrueTorques(module, guidCmdData, vehicleConfigIn)]*3
+
+    # compare the module results to the truth values
+    accuracy = 1e-12
+    testFailCount, testMessages = unitTestSupport.compareArray(trueVector, dataLog.torqueRequestBody, accuracy,
+                                                               "torqueRequestBody", testFailCount, testMessages)
+
+    snippentName = "passFail" + str(setExtTorque)
+    if testFailCount == 0:
+        colorText = 'ForestGreen'
+        print("PASSED: " + module.ModelTag)
+        passedText = r'\textcolor{' + colorText + '}{' + "PASSED" + '}'
+    else:
+        colorText = 'Red'
+        print("Failed: " + module.ModelTag)
+        passedText = r'\textcolor{' + colorText + '}{' + "Failed" + '}'
+
+    # return fail count and join into a single string all messages in the list
+    # testMessage
+    return [testFailCount, ''.join(testMessages)]
```

Comment:
:small_red_triangle_down: Let's refactor the test (in a later commit) to no longer use the testFailCount pattern. Rather just assertions (np.testing etc.) and no calling a first function to only then call a second test function.

Reply:
Done